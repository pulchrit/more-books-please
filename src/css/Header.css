.header {
    background-image: url('books_transparent.png'); 
    background-size: cover;
    background-repeat: no-repeat;
    border-bottom: 1px solid  rgba(10, 34, 73, 0.8);
    border-top: 1px solid  rgba(10, 34, 73, 0.8);
    width: 100%;
    height: 250px;
    display: flex;
    align-items: center;
}

.headline {
    padding: 20px;
    text-align: center;
    color: white;
    background-color: rgba(10, 34, 73, 0.9); 
    width: 100%;
    font-family: 'Love Ya Like A Sister', cursive;
    letter-spacing: 2px;
}