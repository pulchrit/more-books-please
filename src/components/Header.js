import React from 'react';
import { Link } from 'react-router-dom'
import '../css/Header.css';

export default function Header() {
    return (    
        <header className="header">
            <Link to="/" className="headline">More Books, Please</Link>
        </header>
    );
}