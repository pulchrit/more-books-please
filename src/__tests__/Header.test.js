import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Header from '../components/Header';

describe ("Header", () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Header /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <Header />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

})


