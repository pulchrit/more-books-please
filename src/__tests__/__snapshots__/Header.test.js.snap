// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders UI as expected 1`] = `
<header
  className="header"
>
  <h1
    className="headline"
  >
    More Books, Please
  </h1>
</header>
`;
